frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -410,6 +416,21 @@ const getRolloutStatus = (
   return notFailedOrCancelled && previous && previous.pods.length > 0;
 };
 
+const deploymentIsInProgress = (resource: K8sResourceKind): boolean => {
+  return (
+    [
+      DEPLOYMENT_PHASE.new,
+      DEPLOYMENT_PHASE.pending,
+      DEPLOYMENT_PHASE.running,
+      DEPLOYMENT_PHASE.complete,
```

Comment:
Why is `complete` considered in progress?

Reply:
Because we are looking for all phases except for failed or canceled and choose the first two from that filtered array i.e. current and previous. Seems like method name is misleading changed it to `isDeploymentInProgressOrCompleted`.